src/useApolloNetworkStatusReducer.ts

```diff
@@ -22,12 +22,11 @@ export default function useApolloNetworkStatusReducer<T>(
     );
   }
 
-  // Theoretically we should `useEffect` here. However react-apollo@<=2.5.5 uses
-  // React classes and queries can begin to fire in `componentDidMount`.
-  // Therefore we'd miss the request event of an operation. Once `react-apollo`
-  // has migrated, we could bump the peer dependency and use the correct hook
-  // here for a slight performance improvement.
-  useLayoutEffect(() => {
+  // Effects fire bottom-up. Therefore it's possible that when a query is nested
+  // further down the tree than a component using this hook and the query fires
+  // on the initial render, we'll miss the request event. Note that this isn't
+  // an issue when pre-rendering or -fetching on the server side.
```

Comment:
There's theoretically a fix for this, but it increases complexity a bit. Seems like this has never been an issue so far and if it comes up, we can still fix it in a patch release.

Reply:
Actually here's the request for this: https://github.com/molindo/react-apollo-network-status/issues/19#issue-513171800. It shouldn't block this PR IMO though, as the situation is the same before and after this PR.